test/fw/bin/pbs_snapshot

```diff
@@ -189,16 +274,59 @@ if __name__ == '__main__':
         # find the parent directory of the snapshot
         # This will be used to store the map file
         out_abspath = os.path.abspath(out_dir)
-        map_file = os.path.join(out_abspath, "obfuscate.map")
-
-    # Capture the snapshot
-    with PBSSnapUtils(out_dir, server_host=server_host, acct_logs=acct_logs,
-                      daemon_logs=daemon_logs,
-                      additional_hosts=additional_hosts, map_file=map_file,
-                      anonymize=anonymize,
-                      create_tar=create_tar, log_path=log_path,
-                      with_sudo=with_sudo) as snap_utils:
-        outtar = snap_utils.capture_all()
+        if map_file is None:
+            map_file = os.path.join(out_abspath, "obfuscate.map")
+
+    if additional_hosts is not None:
+        du = DshUtils()
+
+        # Run child pbs_snapshot commands on those hosts
+        hostnames = additional_hosts.split(",")
+        timestamp = str(int(time.time()))
+        childsnap_threads = {}
+        for host in hostnames:
+            thread = Thread(target=childsnap_thread, args=(
```

Comment:
that's interesting, I didn;t realize that i was not passing in timestamp, I need to test this out, thanks for pointing it out

Reply:
Just curious, what was your finding @agrawalravi90?

Reply:
My finding was that threading functions cannot access function local variables of the functions where they are invoked from, but they can access global variables, AND, and this is a lil weird, they can also access variables which are defined inside the if __name__ == '__main__' block. since all of the variables that i was passing the threading function were defined inside the if __name__ == '__main__' block, the thread could access them.